File: 937-reorder-data-in-log-files/937-reorder-data-in-log-files.cpp
class Solution {
public:
    vector<string> reorderLogFiles(vector<string>& logs) {
        int n = logs.size();
        vector<pair<string, string>> reorderedFile;
        
        for(int i=0;i<n;i++) {
            string s = logs[i];
            string temp = "";
            int j = 0;
            while(s[j] != ' ') {
                temp += s[j];
                s.erase(s.begin());
            }
            temp += s[j];
            s.erase(s.begin());
            
            if(isalpha(s[0]))
                reorderedFile.push_back({s, temp});
        }
        
        sort(reorderedFile.begin(), reorderedFile.end());
        vector<string> ans;
        
        for(int i=0;i<reorderedFile.size();i++)
            ans.push_back(reorderedFile[i].second + reorderedFile[i].first);
        
        for(int i=0;i<n;i++) {
            string s = logs[i];
            int j = 0;
            while(s[j] != ' ')
                s.erase(s.begin());
            s.erase(s.begin());
            
            if(isdigit(s[0]))
                ans.push_back(logs[i]);
        }
        
        return ans;
    }
};

File: 937-reorder-data-in-log-files/937-reorder-data-in-log-files.java
class Solution {
    public String[] reorderLogFiles(String[] logs) {
        Comparator<String> comp = new Comparator<>() {
            @Override
            public int compare(String log1, String log2) {
                String[] split1 = log1.split(" ", 2);
                String[] split2 = log2.split(" ", 2);
                
                boolean isDigit1 = Character.isDigit(split1[1].charAt(0));
                boolean isDigit2 = Character.isDigit(split2[1].charAt(0));
                
                if(!isDigit1 && !isDigit2) {
                    int cmp = split1[1].compareTo(split2[1]);
                    if(cmp != 0)
                        return cmp;
                    return split1[0].compareTo(split2[0]);
                }
                if(!isDigit1 && isDigit2)
                    return -1;
                else if(isDigit1 && !isDigit2)
                    return 1;
                return 0;
            }
        };
        Arrays.sort(logs, comp);
        return logs;
    }
}